SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DormancyTests/RefreshActorDormancyTest.cpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "RefreshActorDormancyTest.h"
+
+#include "Engine/NetDriver.h"
+#include "Engine/NetworkObjectList.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Net/UnrealNetwork.h"
+#include "RefreshActorDormancyTestActor.h"
+
+ARefreshActorDormancyTest::ARefreshActorDormancyTest()
+{
+	Author = TEXT("Ajanth Sutharsan");
+}
+
+void ARefreshActorDormancyTest::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+
+	DOREPLIFETIME(ThisClass, DormantToAwakeActor);
+	DOREPLIFETIME(ThisClass, AwakeToDormantActor);
+}
+
+void ARefreshActorDormancyTest::PrepareTest()
+{
+	Super::PrepareTest();
+
+	// Step 1 - Spawn dormancy actor which is non dormant and change NetDormancy to DORM_DormantAll after a tick
+	AddStep(TEXT("ServerSpawnNonDormatActor"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		DormantToAwakeActor = SpawnActor<ARefreshActorDormancyTestActor>(FActorSpawnParameters(), /*bRegisterAsAutoDestroy*/ true);
+		AwakeToDormantActor = SpawnActor<ARefreshActorDormancyTestActor>(FActorSpawnParameters(), /*bRegisterAsAutoDestroy*/ true);
+		DormantToAwakeActor->SetupForTest(true);
```

Comment:
This function as a little obtuse to anyone reading the code. Perhaps change it to something like `SetInitiallyDormant`